### PR TITLE
Search filters on mobile

### PR DESCRIPTION
**Pull request by leoshatrushin**
Fix for feature request #7650 by adding a funnel icon to the search bar which opens up a modal were the filters can be set

Note: I don't know how to test the search functionality, so I can't ensure I haven't messed it up

Possible points of discussion:
- The specific funnel icon used
- Whether the funnel icon should disappear for large screen sizes

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1205826751955553) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/search/SearchFilters.tsx
Comment (oetherington) [inline packages/lesswrong/components/search/SearchFilters.tsx:10]:
```suggestion
  ElasticSorting,
```

Mis-matched indentation here

**Review by oetherington — CHANGES_REQUESTED** [review]
Thanks - looks really good! I've left a couple of small suggestion comments on the PR.

WRT the icon, normally the EA Forum uses Heroicons and LessWrong uses MUI icons. It looks like the custom SVG you added is actually the filter icon from Heroicons, so we can just
```
import FunnelIcon from "@heroicons/react/24/outline/FunnelIcon";
```
instead of adding the custom SVG. The MUI filter icon looks pretty bad in my opinion, so I'd probably just stick with heroicons for both sites unless LW object.

FYI @agnesstenlund and @b0b3rt 

Screenshot 2023-10-30 at 17 06 08
Screenshot 2023-10-30 at 17 06 10
Screenshot 2023-10-30 at 17 05 56
Screenshot 2023-10-30 at 17 05 58
Screenshot 2023-10-30 at 17 06 23

File: packages/lesswrong/components/search/SearchPageTabbed.tsx
Comment (b0b3rt) [inline packages/lesswrong/components/search/SearchPageTabbed.tsx:375]: Rather than apply inline styles, we use `classNames` to conditionally apply styles via className.  Search the codebase for `classNames(` for examples of how this works - it'd be something like `classNames(classes.filtersModal, { [classes.modalOpen]: modalOpen, [classes.modalClosed]: !modalOpen })` (and you'd need to define `modalOpen` and `modalClosed` in the `styles` for this component with `display: 'flex'` and `display: 'none'` respectively).

**Comment by b0b3rt** [comment]
Appreciate the fix, and I'm fine with the icon - the MUI icon is indeed not amazing.  Looks good to me, pending addressing comments.

**Comment by darkruby501** [comment]
I tested it quickly and seems to work, regarding icons, yes, LW would use the following or similar:

Material Icons - Material UI 2023-10-31 12-25-01

**Comment by leoshatrushin** [comment]
Fixed the comments about using the palette and inline styles.

I thought I looked for the heroicon and it wasn't there, but I've now removed the custom svg and changed to use the existing icon for the EA forum.

The mui icon however, indeed doesn't seem to be there, so I used custom svg for that. @darkruby501 this is what it looks like
Screenshot 2023-11-02 at 5 32 35 pm

Does the alignment forum typically follow LW choices?

**Comment by oetherington** [comment]
Looks good to me - thanks @leoshatrushin!

Unfortunately, we're stuck on an ancient version of MUI which is why you couldn't find the icon there.

I'll approve, but wait for LW to confirm they're happy with the icon before merging.

**Comment by jpaddison3** [comment]
(For avoidance of doubt: FYI Leo cannot merge this as they don't have write access to this repo, so one of us (either @oetherington or the approving LW dev) will need to click merge.

**Comment by b0b3rt** [comment]
Ruby's current offline - custom SVG looks good to me.  Double-checking that it looks correct in dark mode as well?  If so, all clear from the LW side.

**Pull request merged**
